js/request.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+import * as headers from "./headers";
+import * as body from "./body";
+
+import * as streams from "@stardazed/streams";
+
+import * as domTypes from "./dom_types";
+
+const { Headers } = headers;
+const { ReadableStream } = streams;
+
+function byteUpperCase(s: string): string {
+  return String(s).replace(/[a-z]/g, function byteUpperCaseReplace(c): string {
+    return c.toUpperCase();
+  });
+}
+
+function normalizeMethod(m: string): string {
+  const u = byteUpperCase(m);
+  if (
+    u === "DELETE" ||
+    u === "GET" ||
+    u === "HEAD" ||
+    u === "OPTIONS" ||
+    u === "POST" ||
+    u === "PUT"
+  ) {
+    return u;
+  }
+  return m;
+}
+
+/**
+ * An HTTP request
+ * @param {Blob|String} [body]
+ * @param {Object} [init]
+ */
+export class Request extends body.Body implements domTypes.Request {
+  public method: string;
+  public url: string;
+  public credentials?: "omit" | "same-origin" | "include";
+  public headers: domTypes.Headers;
+
+  constructor(input: domTypes.RequestInfo, init?: domTypes.RequestInit) {
```

Comment:
why not:
`constructor(input: domTypes.RequestInfo, init: domTypes.RequestInit = {}) {`
as 
`if (!init) {
      init = {};
    }`
follows?